source/common/runtime/runtime_features.cc

```diff
@@ -35,6 +35,7 @@ constexpr const char* runtime_features[] = {
     "envoy.reloadable_features.new_http2_connection_pool_behavior",
     "envoy.deprecated_features.allow_deprecated_extension_names",
     "envoy.reloadable_features.ext_authz_http_service_enable_case_sensitive_string_matcher",
+    "envoy.reloadable_features.new_codec_behavior",
```

Comment:
I'm inclined to think we should version history document this in this PR - that we're introducing due to upcoming high risk refactors, and you can flip it if you want to opt out, blah blah blah.

Reply:
Done!